library/oneview_event.py

```diff
@@ -0,0 +1,129 @@
+short_description: Manage OneView Events.
+description:
+    - Provides an interface to manage Events. Can only create.
+version_added: "2.3"
+requirements:
+    - "python >= 2.7.9"
+    - "hpOneView >= 3.1.0"
+author: "Felipe Bulsoni (@fgbulsoni)"
+options:
+    state:
+        description:
+            - Indicates the desired state for the Event.
+              C(present) will ensure data properties are compliant with OneView. This operation is non-idempotent.
+        choices: ['present']
+    data:
+        description:
+            - List with the Event properties.
+        required: true
+
+extends_documentation_fragment:
+    - oneview
+'''
+
+EXAMPLES = '''
+- name: Ensure that the Event is present using a test type id
+  oneview_event:
+    config: "{{ config_file_path }}"
+    state: present
+    data:
+      description: "This is a very simple test event"
+      eventTypeID: "hp.justATest"
+      eventDetails: [{
+                      eventItemName: "ipv4Address",
+                      eventItemValue: "198.51.100.5",
+                      isThisVarbindData: "false",
+                      varBindOrderIndex: -1
+                      }]
+'''
+
+RETURN = '''
+event:
+    description: Has the facts about the OneView Events.
+    returned: On state 'present'. Can be null.
+    type: complex
+'''
+
+from ansible.module_utils.basic import AnsibleModule
+from module_utils.oneview import OneViewModuleBase
+
+
+class EventModule(OneViewModuleBase):
+    MSG_CREATED = 'Event created successfully.'
+    RESOURCE_FACT_NAME = 'event'
+
+    def __init__(self):
+
+        additional_arg_spec = dict(data=dict(required=True, type='dict'),
+                                   state=dict(
+                                       required=True,
+                                       choices=['present']))
+
+        super(EventModule, self).__init__(additional_arg_spec=additional_arg_spec)
+
+        self.resource_client = self.oneview_client.events
+
+    def resource_present(self, resource, fact_name):
+        """
+        Implementation of the present state for the OneView resource 'Event'.
+        It creates the resource. This operation is not idempotent as there is
+        no way to query for a resource other than its uri.
+
+        Args:
+            resource (dict):
+                Resource to create.
+            fact_name (str):
+                Name of the fact returned to the Ansible.
+
+        Returns:
+            A dictionary with the expected arguments for the event
+        """
+        changed = False
+        resource = self.resource_client.create(self.data)
+        msg = self.MSG_CREATED
+        changed = True
+
+        return dict(
+            msg=msg,
+            changed=changed,
+            ansible_facts={fact_name: resource}
+        )
+
+    def execute_module(self):
+        resource = None
+
+        if self.state == 'present':
+            return self.resource_present(resource, self.RESOURCE_FACT_NAME)
```

Comment:
I would replace this line (and the entire method above) by:
```python
resource = self.resource_client.create(self.data)
return dict(changed=True, msg=MSG_CREATED, ansible_facts=dict(event=resource))
```
That's my opinion, sometimes less is more. Feel free to do it or not. ;)